src/distances/strings.rs

```diff
@@ -43,9 +44,40 @@ pub fn levenshtein<T: Number, U: Number>(x: &[T], y: &[T]) -> U {
     }
 }
 
+// Wrapper function for calculating distance based on NW-aligned sequences (returns edit distance only)
+pub fn needleman_wunsch<T: Number, U: Number>(x: &[T], y: &[T]) -> U {
+    let table = alignment_helpers::compute_nw_table(x, y);
+    let edit_distance: usize = table[table.len() - 1][table[0].len() - 1].0;
+
+    U::from(edit_distance).unwrap()
+}
+
+/*
+Function for calculating calculating distance based on NW-aligned sequences (returns edit
+distance AND alignment associated with shortest edit distance)
+
+For now, in cases where there exist ties for the shortest edit distance, we only return
+one alignment
+*/
+pub fn needleman_wunsch_with_edits<T: Number, U: Number>(
+    x: &[T],
+    y: &[T],
+) -> ((Vec<alignment_helpers::Edit<T>>, Vec<alignment_helpers::Edit<T>>), U) {
+    let table = alignment_helpers::compute_nw_table(x, y);
+    let (aligned_x, aligned_y) = alignment_helpers::traceback_recursive(&table, (x, y));
+
+    let edit_x_into_y = alignment_helpers::alignment_to_edits(&aligned_x, &aligned_y);
+    let edit_y_into_x = alignment_helpers::alignment_to_edits(&aligned_y, &aligned_x);
+
+    let edit_distance: usize = table[table.len() - 1][table[0].len() - 1].0;
+
+    ((edit_x_into_y, edit_y_into_x), U::from(edit_distance).unwrap())
+}
```

Comment:
Add recursive and iterative variants of this, changing only line 67.

Reply:
Okay. I assume you mean with some sort of input flag like "recursive = TRUE"

Reply:
For now, I just want two separate functions. This will be for benchmarks when you make the `alignment_helpers` module `pub(crate)`